components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -0,0 +1,569 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! This module contains batch split related processing logic.
+//!
+//! Process Overview
+//!
+//! Propose:
+//! - Nothing special except for validating batch split requests (ex: split keys
+//!   are in ascending order).
+//!
+//! Execution:
+//! - exec_batch_split: Create and initialize metapb::region for split regions
+//!   and derived regions. Then, create checkpoints of the current talbet for
+//!   split regions and derived region to make tablet physical isolated. Update
+//!   the parent region's region state without persistency. Send the new regions
+//!   (including derived region) back to raftstore.
+//!
+//! Result apply:
+//! - todo
+//!
+//! Split peer creation and initlization:
+//! - todo
+//!
+//! Split finish:
+//! - todo
+
+use std::collections::VecDeque;
+
+use engine_traits::{
+    Checkpointer, KvEngine, OpenOptions, RaftEngine, TabletFactory, CF_DEFAULT, SPLIT_PREFIX,
+};
+use kvproto::{
+    metapb::Region,
+    raft_cmdpb::{AdminRequest, AdminResponse, RaftCmdRequest, SplitRequest},
+    raft_serverpb::RegionLocalState,
+};
+use protobuf::Message;
+use raftstore::{
+    coprocessor::split_observer::{is_valid_split_key, strip_timestamp_if_exists},
+    store::{
+        fsm::apply::{extract_split_keys, validate_batch_split},
+        metrics::PEER_ADMIN_CMD_COUNTER,
+        util::{self, KeysInfoFormatter},
+        PeerStat, ProposalContext, RAFT_INIT_LOG_INDEX,
+    },
+    Result,
+};
+use slog::{info, warn, Logger};
+use tikv_util::box_err;
+
+use crate::{
+    batch::StoreContext,
+    fsm::ApplyResReporter,
+    operation::AdminCmdResult,
+    raft::{Apply, Peer},
+    router::ApplyRes,
+};
+
+#[derive(Debug)]
+pub struct SplitResult {
+    pub regions: Vec<Region>,
+    pub derived: Region,
```

Comment:
Instead of returning the a duplicated region, I think an index to show which region in `regions` is derived should be sufficient.

Reply:
Done